intro-to-object-oriented-programming/what-is-oop.md

```diff
@@ -0,0 +1,162 @@
+
+Objects are _things_ that can be their own specific data types, hold unique data, and can do unique behavior.
+
+For example, Python, the programming language, demonstrates using OOP to solve problems.
+
+Numbers, text, sequences, maps, sets, and the idea of true and false can be used to solve problems. Python captures each of these ideas into different data types. Python has defined the data types `int`, `float`, `str`, `dict`, `set`, `bool`, and more.
+
+Each of these data types:
+
+- represent a different, unique idea as data
+- have unique behavior that is meaningful to itself
+
+For example...
+
+- Integers
+  - represent whole numbers
+  - can be subtracted with another integer or float
+- Strings
+  - represent text
+  - can be concatenated with other strings
+- Booleans
+  - represent true or false
+  - can be negated
+
+### Defining Classes
+
+Beyond using Python's built-in data types, we can define our own data types.
+
+To define our own data type, we create a **class.** A **class** is code that defines a template for creating similar objects.
+
+Each class definition will have:
+
+- A class name
+  - This name is used as the "data type" name
+- Generic **attributes**
+  - An attribute is a piece of data
+  - A class definition holding an attribute means we expect this class to hold a specific piece of data
+- Generic **methods**
+  - A method is a function that describes behavior or action
+- Logic to describe what happens every time the class is used to make an **instance.**
+
+### !callout-info
+
+## Examples: Idea to Class
+
+Observe these examples of classes that already exist, or classes that could exist.
+
+| Idea to represent                                                                                         | Class Name |
+| --------------------------------------------------------------------------------------------------------- | ---------- |
+| A class that describes the idea of a sequence of text                                                     | String     |
+| A class that describes an ordered list                                                                    | List       |
+| A class that describes a vehicle driver working at a ride-share app, who has a ride history and a vehicle | Driver     |
+| A class that describes a music album sold in a digital music store, and all its data                      | Album      |
+
+### !end-callout
+
+### Creating Instances of Classes
+
+An **instance** is a concrete, literal manifestation of a class.
+
+A _class_ defines a data type _generically_, much like a template, or blueprint. It determines what that data type _can potentially_ hold as state or behavior.
+
+On the other hand, an _instance_ of a class is a _concrete thing_ of that data type, with concrete state and behavior. One instance is one specific thing that has literal state and literal behavior. Each instance allocates and uses memory, just like other pieces of data.
```

Comment:
I need a second opinion here but generic vs. concrete/literal state or behavior is hard (at least for me) to grasp. Can we simplify this, perhaps? Maybe something along the lines of:

> A _class_ defines a general data type, much like a template, blueprint, or recipe. It determines the initial state or behavior of that data type. 
> 
> On the other hand, an _instance_  is the implementation of a class. The class provides instructions to create a cookie and the instance would be the actual cookie (in its physical and yummy form)! The instance will be like other pieces of data that allocate and use memory and have concrete state and behavior. 

We would probably extend this 'cookie' analogy to the next paragraph. 

@tildeee